### PR TITLE
Allow snapshots to be loaded off disk rather than read into ram first

### PR DESCRIPTION
**Pull request by Vagabond**
This change should help devices with less memory load larger snapshots.

### CODE REVIEW

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 1]
```diff
@@ -28,11 +28,11 @@
          diff/2
         ]).
 
--ifdef(TEST).
+%-ifdef(TEST).
```
Comment (jadeallenx) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:31]: If this should be exported, just export it?

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 2]
```diff
@@ -610,18 +623,28 @@ load_blocks(Ledger0, Chain, Snapshot) ->
     print_memory(),
```
Comment (jadeallenx) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:623]: Hopefully this print function is conditional or routed through lager log levels.

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl (hunk 1 above)
Comment (Vagabond) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:31]: This was a side effect of my shell testing, it does not need to be exported.

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 3]
```diff
@@ -1444,6 +1552,19 @@ mk_bin_iterator(<<SizK:32/little-unsigned-integer, K:SizK/binary,
...
 
+mk_file_iterator(_FD, End, End) ->
+    fun() -> ok end;
+mk_file_iterator(FD, Pos, End) when Pos < End ->
+    fun() ->
+            {ok, Pos} = file:position(FD, {bof, Pos}),
+            {ok, <<SizK:32/integer-unsigned-little>>} = file:read(FD, 4),
+            {ok, <<K:SizK/binary, SizV:32/integer-unsigned-little>>} = file:read(FD, SizK + 4),
+            {ok, V} = file:read(FD, SizV),
+            io:format("read key of size ~p and value of size ~p", [SizK, SizV]),
```
Comment (evanmcc) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:1563]: remove

**Review by evanmcc — APPROVED** [review]

**Pull request merged**